### PR TITLE
Remove the corresponding measure from Taffy when a `CalculatedSize` component is removed.

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

When a `CalculatedSize` component from a UI Node entity is removed, the corresponding Taffy measure isn't removed which will mess up the layout in confusing, unpredictable ways.

## Solution

Iterate through all the entities with removed `CalculatedSize` components and remove the corresponding Taffy measures.

### CODE REVIEW

File: crates/bevy_ui/src/flex/mod.rs [hunk 1]
```diff
@@ -320,6 +328,11 @@ pub fn flex_node_system(
     // clean up removed nodes
     flex_surface.remove_entities(removed_nodes.iter());
 
+    // When a `CalculatedSize` component is removed from an entity, we need to remove the measure from the corresponding taffy node.
+    for entity in removed_calculated_sizes.iter() {
```
Comment (TimJentzsch) [inline crates/bevy_ui/src/flex/mod.rs:332]: What would you think about moving this into a separate system?

This one seems to be getting quite big and doing a lot of different things.

**Comment by TimJentzsch** [comment]
How challenging would it be to add a test for this?
I think that would be very nice to have to avoid regressions in the future

File: crates/bevy_ui/src/flex/mod.rs (hunk 1 above)
Comment (ickshonpe) [reply crates/bevy_ui/src/flex/mod.rs:332]: Definitely needs to be done, and I found this bug while working on some other changes that require `flex_node_system` to be split up. I'm not quite sure how they'll work out yet though, so I just put in this PR that fixes the bug with minimal changes etc.

**Comment by ickshonpe** [comment]
> How challenging would it be to add a test for this? I think that would be very nice to have to avoid regressions in the future

It's not too difficult, but `flex_node_system` needs to be split up so we can run just the layout algorithms without a window.

**Review by TimJentzsch — APPROVED** [review]

**Review by cart — APPROVED** [review]

**Pull request merged**